File: server/internal/model/health.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
	"gorm.io/gorm"
)

type HealthRecord struct {
	ID          uuid.UUID      `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	PetID       uuid.UUID      `gorm:"type:uuid;index;not null"`
	Type        string         `gorm:"size:30;not null"`
	Title       string         `gorm:"size:200;not null"`
	Description string         `gorm:"type:text"`
	Data        datatypes.JSON `gorm:"type:jsonb"`
	DueDate     *time.Time
	ProviderID  *uuid.UUID     `gorm:"type:uuid"`
	Attachments datatypes.JSON `gorm:"type:jsonb"`
	RecordedAt  time.Time      `gorm:"not null"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
	DeletedAt   gorm.DeletedAt `gorm:"index"`
	Pet         Pet            `gorm:"foreignKey:PetID"`
}

func (h *HealthRecord) BeforeCreate(_ *gorm.DB) error {
	if h.ID == uuid.Nil {
		h.ID = uuid.New()
	}
	return nil
}

type HealthAlert struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	PetID       uuid.UUID `gorm:"type:uuid;index;not null"`
	AlertType   string    `gorm:"size:30;not null"`
	Severity    string    `gorm:"size:10;not null"`
	Title       string    `gorm:"size:200;not null"`
	Message     string    `gorm:"type:text;not null"`
	Source      string    `gorm:"size:30"`
	IsRead      bool      `gorm:"default:false"`
	IsDismissed bool      `gorm:"default:false"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
	DeletedAt   gorm.DeletedAt `gorm:"index"`
	Pet         Pet            `gorm:"foreignKey:PetID"`
}

func (h *HealthAlert) BeforeCreate(_ *gorm.DB) error {
	if h.ID == uuid.Nil {
		h.ID = uuid.New()
	}
	return nil
}

File: apps/mobile/types/health.ts
export interface HealthRecord {
  id: string;
  pet_id: string;
  type: string;
  title: string;
  description: string;
  data: Record<string, unknown>;
  due_date?: string;
  attachments: string[];
  recorded_at: string;
  created_at: string;
  updated_at: string;
}

export interface HealthAlert {
  id: string;
  pet_id: string;
  alert_type: string;
  severity: string;
  title: string;
  message: string;
  source: string;
  is_read: boolean;
  is_dismissed: boolean;
  created_at: string;
}

export interface HealthSummary {
  score: number;
  status: "excellent" | "stable" | "watch" | "critical" | string;
  insights: string[];
  recommended_actions: string[];
  data_points_analyzed: number;
  generated_at: string;
}

export interface HealthAIAnswer {
  question: string;
  answer: string;
  created_at: string;
}

export interface CreateHealthRecordPayload {
  type: string;
  title: string;
  description?: string;
  data?: Record<string, unknown>;
  due_date?: string;
  attachments?: string[];
  recorded_at?: string;
}
